backend/algorithms/objects/conditions.py

```diff
@@ -239,7 +249,20 @@ def _validate_course(category: Category):
                 unlocked, warnings = list(zip(*validations))
                 satisfied = all(unlocked) if category.logic == Logic.AND else any(unlocked)
                 return satisfied, warnings
-            elif isinstance(category, ClassCategory):
+
+
+            course = course.class_name
```

Comment:
in this scope, for `course = course.class_name`, where is the `course` object on the RHS of the equals coming from?
afaict it may not be initialised at this point

Reply:
tbh i don't think I changed much in the GradeCondition section except adding a string to one of the returns so will look through this now